spring/boot-autoconfigure/src/main/java/com/linecorp/armeria/spring/ArmeriaBeanPostProcessor.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.spring;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+
+import java.beans.PropertyDescriptor;
+import java.lang.reflect.AnnotatedElement;
+import java.lang.reflect.Member;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.annotation.Nullable;
+
+import org.springframework.beans.BeanUtils;
+import org.springframework.beans.PropertyValues;
+import org.springframework.beans.factory.BeanCreationException;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.annotation.InjectionMetadata;
+import org.springframework.beans.factory.annotation.InjectionMetadata.InjectedElement;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.beans.factory.config.InstantiationAwareBeanPostProcessor;
+import org.springframework.core.BridgeMethodResolver;
+import org.springframework.core.Ordered;
+import org.springframework.util.ReflectionUtils;
+import org.springframework.util.StringUtils;
+
+import com.linecorp.armeria.common.SessionProtocol;
+import com.linecorp.armeria.server.Server;
+
+/**
+ * {@link BeanPostProcessor} implementation that autowires annotated fields, setter methods, and arbitrary
+ * config methods. Such members to be injected are detected through annotations:
+ * by default, {@link LocalArmeriaPort} and {@link LocalArmeriaPorts} annotations.
+ *
+ * <p>This class was created by referring to classes implementing
+ * Spring's {@link InstantiationAwareBeanPostProcessor}.
+ */
+public final class ArmeriaBeanPostProcessor implements InstantiationAwareBeanPostProcessor, Ordered {
+
+    private final Map<String, InjectionMetadata> injectionMetadataCache = new ConcurrentHashMap<>(256);
+
+    private final BeanFactory beanFactory;
+
+    /**
+     * Creates a new instance.
+     */
+    public ArmeriaBeanPostProcessor(BeanFactory beanFactory) {
+        this.beanFactory = requireNonNull(beanFactory, "beanFactory");
+    }
+
+    @Override
+    public int getOrder() {
+        return Ordered.LOWEST_PRECEDENCE - 2;
```

Comment:
Could you leave a comment what `-2` is for?

Reply:
There is no special meaning for order. It just fits the same as `AutowiredAnnotationBeanPostProcessor`.

If I think we don't need an order, it doesn't seem like a way to specify it. What do you think?

Reply:
Could you add an in-code comment that explains this? Not a Github comment

Reply:
What do you think about not implementing `Ordered`? When I looked at some `BeanPostProcessor` implementations, I saw no implementation.

Reply:
> What do you think about not implementing Ordered?

+1. If we need to, I think we can implement it later.

Reply:
Removed. be30ae3